### ISSUE TITLE
Prefer String::equalsToIgnoreCase - "foo".equals("Foo".toLowercase()) -> "foo".equalsToIgnoreCase("Foo")

### ISSUE THREAD

**Issue by timo-abele** [openrewrite/rewrite-static-analysis#274]
## What problem are you trying to solve?

Just came across this finding from IntelliJ:
```java
String upperFoo = "FoO"
"foo".equals(upperFoo.toLowercase())
```
can be
```java
String upperFoo = "FoO"
"foo".equalsToIgnoreCase(upperFoo)
```

Would be nice to have that automated.
I've considered posting this in rewrite-migrate-java, but as there is no "since" in the [API](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#equalsIgnoreCase(java.lang.String)) I guess it has been around forever and was never a new language feature.

**Comment by knutwannheden** [comment]
Correct. There is in fact a `StringRules` recipe. This was intended as a demo for Refaster recipes, but there is a recipe in there which does almost what you want. In fact that recipe could be simplified.

I suspect a similar Refaster rule also exists in Errorprone Support, which is available via `rewrite-third-party`.

**Comment by Stephan202** [comment]
> I suspect a similar Refaster rule also exists in Errorprone Support, which is available via `rewrite-third-party`.

I had a look; we don't have such a rule. I'm also not sure we would accept such a contribution, as we instead prefer that a `toLowerCase()` is rewritten to accept a locale. The [`StringCaseLocaleUsage`](https://errorprone.info/bugpattern/StringCaseLocaleUsage) check provided by Error Prone was in fact contributed by Picnic. See also [this SO thread](https://stackoverflow.com/questions/43988852/case-insensitive-comparisons-across-locales-in-java) about caveats around the use of `String#equalsToIgnoreCase`; I'd generally try to stay away from this method.

**Comment by knutwannheden** [comment]
@Stephan202 Interesting. I was not aware of this issue.

**Comment by timo-abele** [comment]
> See also [this SO thread](https://stackoverflow.com/questions/43988852/case-insensitive-comparisons-across-locales-in-java) about caveats around the use of String#equalsToIgnoreCase; I'd generally try to stay away from this method.

Interesting. I ended up not following the IntelliJ suggestion either. I'm not mad if the issue is rejected.

**Comment by knutwannheden** [comment]
As implemented in the demo recipe it should be safe: https://github.com/openrewrite/rewrite-migrate-java/blob/582216517bdbc601ec0203f87bbbbb8e8a424f22/src/main/java/org/openrewrite/java/migrate/lang/StringRules.java#L77

**Comment by knutwannheden** [comment]
... or maybe not. If the case conversions are known to use certain locales like `ROOT` I suppose it would be a safe operation.
